pyvista/core/filters/image_data.py

```diff
@@ -978,7 +979,13 @@ def contour_labeled(
         _vtk.vtkLogger.SetStderrVerbosity(verbosity)
         return cast(pyvista.PolyData, wrap(alg.GetOutput()))
 
-    def points_to_cells(self, scalars: str | None = None, *, copy: bool = True):
+    def points_to_cells(
+        self,
+        scalars: str | None = None,
+        *,
+        resize_dims: bool | VectorLike[bool] | Literal['1D', '2D', '3D'] = True,
```

Comment:
@MatthewFlamm Looking for some additional feedback on the default behaviour here. With this line it's currently set to resize inputs, i.e. 1D and 2D inputs become 3D by default. But, this breaks backwards compatibility, since `points_to_cells` (and `pad_image`) were both originally written _not_ resize inputs. So is it preferred to:

1. break backwards compatibility and make this filter resize all dimensions by default
2. keep the original behaviour and only resize non-singleton dimensions

The original "only resize non-singleton" dimensions behaviour makes sense to me for working with actual 2D or 3D images. But I don't think most filters in pyvista were written with 2D image cases specifically in mind with special handling like this. So, what to do? I am in favour of whichever API produces the more consistent and expected "pyvista-like" experience. Based on the historical naming of `ImageData` as `UniformGrid` this class may have been less geared towards working with images specifically...

Reply:
For context, I set the default to `True` because it's the less likely-to-fail option.

The original "only resize non-singleton" dimensions are not exactly achievable with the current state of the PR, but the same intended functionality (i.e., working with 1D and 2D images)  can be obtained, IMO, with the literals `'1D'` and `'2D'`. 

We could add a `'non-singleton'` literal as a fourth option to maintain the previous behaviour, but I am biased against it for two reasons:
- the expected result is not explicit.
- as long the user knows what are the input dims and what are the required output dims, it does not bring extra functionality.

But I would still be happy to have it if you think it's useful :wink:

Reply:
I like specifying the dimensionality with the literals, and am fine with the changes generally. Since these are breaking changes though and I was thinking it might be good to get another opinion on the matter.